### PR TITLE
Image slider feature

### PR DESCRIPTION
**Pull request by koba-wild**
Image Slider + Navigation. Have an issue with dots under images

### CODE REVIEW

File: app/src/main/java/com/example/dragonx/presentation/RocketDetails/ImageSliderAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                val item = getItem(position) as String
+                holder.bind(item)
+            }
+        }
+    }
+
+    class ImageViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val sliderImageView = itemView.findViewById<ShapeableImageView>(R.id.sliderImageView)
+        fun bind(image: String) {
+            val requestOption = RequestOptions()
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketDetails/ImageSliderAdapter.kt:33]: Variable is in use?

File: app/src/main/java/com/example/dragonx/presentation/RocketDetails/RocketDetailsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                dots[i]?.text = Html.fromHtml("&#8226")
+            }
+            dots[i]?.textSize = 38f
+            dots[i]?.setTextColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.white))
+            view.layoutDots.addView(dots[i])
+        }
+        if (dots.isNotEmpty()) {
+            dots[currentPage]?.setTextColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.teal_700))
+        }
+    }
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketDetails/RocketDetailsFragment.kt:77]: Dots are not symbols, they are views.

File: app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.example.dragonx.model.data.RocketList
+import com.example.dragonx.viewmodel.RocketListViewModel
+
+
+class RocketListFragment : Fragment(), OnRocketClickListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListFragment.kt:28]: Method is not used.

File: app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        viewModel.rocketsData.observe(viewLifecycleOwner, {
+            adapter.submitList(it)
+        })
+
+        viewModel.status.observe(viewLifecycleOwner, {
+            when (it) {
+                ApiStatus.ERROR -> {
+                    errorPicture.isVisible = true
+                    errorPicture.setImageResource(R.drawable.ic_connection_error)
+                    Toast.makeText(context, "Exception occurred: ${viewModel.myException}", Toast.LENGTH_LONG).show()
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListFragment.kt:52]: All visible text for user have to be in string.xml

File: app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.example.dragonx.model.util.ApiStatus
+import kotlinx.coroutines.*
+
+class RocketListViewModel : ViewModel() {
+    private val _status = MutableLiveData<ApiStatus>()
+    val status: LiveData<ApiStatus>
+        get() = _status
+    var rocketsData = MutableLiveData<List<RocketList>>()
+    private val rocketListParser = GetRocketList()
+    lateinit var myException: Exception
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListViewModel.kt:15]: What about encapsulation? I advised you to use sealed classes.

File: app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    init {
+        getRockets()
+    }
+
+    fun getRockets() {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO){
+                try {
+                    _status.postValue(ApiStatus.LOADING)
+                    rocketsData.postValue(rocketListParser.getRocketData() ?:null)
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListViewModel.kt:26]: if getRocketData() is null you return null?

File: app/src/main/res/navigation/navigation.xml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    </fragment>
+    <fragment
+        android:id="@+id/rocketDetailsFragment"
+        android:name="com.example.dragonx.presentation.RocketDetails.RocketDetailsFragment"
+        android:label="Rocket details"
+        tools:layout="@layout/fragment_rocket_details" >
+        <argument
+            android:name="rocketNumber"
+            app:argType="integer"
+            android:defaultValue="0" />
```
Comment (EvgenePrytula) [inline app/src/main/res/navigation/navigation.xml:25]: Try to pass entire rocket here.

**Review by EvgenePrytula — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 8]
```diff
@@ -66,6 +69,9 @@ dependencies {
     implementation 'androidx.navigation:navigation-ui-ktx:2.3.5'
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
[the remark this answers is not in the document]
Comment (koba-wild) [reply app/build.gradle:70]: I don't think so. I will delete those dependencies.

File: app/build.gradle [hunk 9]
```diff
@@ -34,20 +37,20 @@ android {
...
-    def lifecycle_version = "2.3.1"
 
-    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$lifecycle_version"
+    implementation 'androidx.lifecycle:lifecycle-livedata-ktx:2.3.1'
+    implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.1'
 
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
     implementation 'androidx.core:core-ktx:1.5.0'
-    implementation 'androidx.appcompat:appcompat:1.3.0'
+    implementation 'androidx.appcompat:appcompat:1.3.1'
```
[the remark this answers is not in the document]
Comment (koba-wild) [reply app/build.gradle:53]: I can't tell you(

File: app/src/main/java/com/example/dragonx/domain/GetRocketList.kt [hunk 10]
```diff
@@ -0,0 +1,25 @@
+package com.example.dragonx.domain
+import com.example.dragonx.NetworkService.Rocket
+import com.example.dragonx.model.util.ApiStatus
+import com.example.dragonx.model.data.RocketList
+
+
+class GetRocketList {
```
Comment (EvgenePrytula) [inline app/src/main/java/com/example/dragonx/domain/GetRocketList.kt:7]: Clean Arch.

**Pull request merged**